src/desktop_notifier/backends/winrt.py

```diff
@@ -60,6 +60,8 @@ def register_hkey(app_id: str, app_name: str) -> None:
     key_path = f"SOFTWARE\\Classes\\AppUserModelId\\{app_id}"
     with winreg.CreateKeyEx(winreg.HKEY_CURRENT_USER, key_path) as master_key:
         winreg.SetValueEx(master_key, "DisplayName", 0, winreg.REG_SZ, app_name)
+        if app_icon is not None:
```

Comment:
```suggestion
        if app_icon is not None and app_icon.is_file():
```

`app_icon.as_uri()` will throw an `AttributeError` if the icon is referenced by name instead of by URI or file path.